### DISCUSSION TITLE
docs: How to use an index property in MongoDb

### DISCUSSION THREAD

**Discussion by RaidelRoss** [mikro-orm/mikro-orm#3735] [Q&A] [answered]
I'm using:
Node: v16.14.2
mikro-orm/core@5.4.1
mikro-orm/mongodb@5.4.1

I am using the following code to create the User entity.

```
@Entity()
export class User{

  @PrimaryKey()
  _id!: ObjectId;

  @SerializedPrimaryKey()
  id!: string; // won't be saved in the database

  @Index({ name: '_idx_score'})
  @Property() score: number = 0;

}
```

I am trying to use `score` as an indexed property. it is being created in the database correctly with the name `_idx_score`

I can't find it in the documentation
-How to use indexing in a query?
-Define if indexing is ascending or descending

There is the documentation that has reference to indexing but does not show how to use it
[index-and-unique ](https://mikro-orm.io/docs/decorators#index-and-unique)
[indexes](https://mikro-orm.io/docs/usage-with-mongo#indexes)

Thank you

**Comment by B4nan** [comment, marked as answer]
> -How to use indexing in a query?

You don't "use indexes", you query the collections, and based on your query indexes are used. 

> Define if indexing is ascending or descending

That is one of the examples in the page you linked:

<img width="827" alt="image" src="https://user-images.githubusercontent.com/615580/201547691-c92881bc-4776-4a43-9d2f-46529a375812.png">

This way you work with the same options parameter that the mongo docs:

https://www.mongodb.com/docs/manual/reference/method/db.collection.createIndex/
